source/common/router/rds_impl.cc

```diff
@@ -206,6 +217,37 @@ void RdsRouteConfigProviderImpl::onConfigUpdate() {
     prev_config->config_ = new_config;
     return previous;
   });
+
+  const auto aliases = config_update_info_->aliasesInLastVhdsUpdate();
+  // Regular (non-VHDS) RDS updates don't populate aliases fields in resources.
+  if (aliases.empty()) {
+    return;
+  }
+
+  // Notifies connections that RouteConfiguration update has been propagated.
+  // Callbacks processing is performed in FIFO order and stops when the alias(es) used in the VHDS
+  // update request do not match the aliases in the update response. The assumption is the response
+  // to a request is not going to arrive before all requests before it have been responded to.
+  config_update_callbacks_->runOnAllThreads(
+      [aliases](ThreadLocal::ThreadLocalObjectSharedPtr previous)
+          -> ThreadLocal::ThreadLocalObjectSharedPtr {
+        auto callbacks = std::dynamic_pointer_cast<ThreadLocalCallbacks>(previous)->callbacks_;
+        std::vector<std::string> aliases_not_in_update;
+        while (!callbacks.empty()) {
+          auto update_on_demand_callback = callbacks.front();
+          std::set_difference(update_on_demand_callback.aliases_.begin(),
+                              update_on_demand_callback.aliases_.end(), aliases.begin(),
+                              aliases.end(), std::back_inserter(aliases_not_in_update));
+          if (aliases_not_in_update.empty()) {
```

Comment:
I think you might be able to copy `aliases_not_in_update.empty()` up into the `while` condition, and get rid of the `else break`. I won't 100% argue that I'm sure it's cleaner, but I think I like it a bit more that way?